tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -401,6 +411,189 @@ def stat(self, filename):
+        """
+        fs, path = self._fs_path(filename)
+
+        mode = "rb" if binary_mode else "r"
+        encoding = None if binary_mode else "utf8"
+        if not exists(filename):
+            raise errors.NotFoundError(
+                None, None, "Not Found: " + compat.as_text(filename)
+            )
+        offset = None
+        if continue_from is not None:
+            if not binary_mode:
+                raise errors.InvalidArgumentError(
+                    None,
+                    None,
+                    "continue_from not supported with fsspec text mode",
+                )
+            offset = continue_from.get("opaque_offset", None)
+        with fs.open(path, mode, encoding=encoding) as f:
+            if offset is not None:
+                f.seek(offset)
+            data = f.read(size)
+            # The new offset may not be `offset + len(data)`, due to decoding
+            # and newline translation.
+            # So, just measure it in whatever terms the underlying stream uses.
+            continuation_token = (
+                {"opaque_offset": f.tell()} if binary_mode else {}
+            )
+            return (data, continuation_token)
+
+    @_translate_errors
+    def write(self, filename, file_content, binary_mode=False):
+        """Writes string file contents to a file.
+
+        Args:
+            filename: string, a path
+            file_content: string, the contents
+            binary_mode: bool, write as binary if True, otherwise text
+        """
+        self._write(filename, file_content, "wb" if binary_mode else "w")
+
+    @_translate_errors
+    def append(self, filename, file_content, binary_mode=False):
+        """Append string file contents to a file.
+
+        Args:
+            filename: string, a path
+            file_content: string, the contents to append
+            binary_mode: bool, write as binary if True, otherwise text
+        """
+        self._write(filename, file_content, "ab" if binary_mode else "a")
+
+    def _write(self, filename, file_content, mode):
+        fs, path = self._fs_path(filename)
+        encoding = None if "b" in mode else "utf8"
+        with fs.open(path, mode, encoding=encoding) as f:
+            compatify = compat.as_bytes if "b" in mode else compat.as_text
+            f.write(compatify(file_content))
+
+    @_translate_errors
+    def glob(self, filename):
+        """Returns a list of files that match the given pattern(s)."""
+        if isinstance(filename, bytes):
+            filename = filename.decode("utf-8")
+        fs, _, _ = fsspec.get_fs_token_paths(
+            filename,
+        )
+        files = fs.glob(filename)
+        SEPARATOR = "://"
+
+        prefixed_files = []
+        if SEPARATOR in filename:
+            for file in files:
+                if SEPARATOR not in file:
+                    file = fs.protocol + SEPARATOR + file
+                prefixed_files.append(file)
+        return prefixed_files
```

Comment:
I'm not certain this will work in all cases... there are chained systems (ex: `dask::s3://bucket/key`) which fsspec supports, local systems where paths don't have a protocol (ex: `C:\Users\reubend\file.txt`) and a few other edge cases.

Have you tested this with those? My initial guess is that putting the protocol and then the separator might break those cases.

Reply:
updated the diff to handle chained file systems and added test cases for them